app/jobs/kit_import_job.rb

```diff
@@ -175,13 +180,23 @@ def import_rules
   end
 
   def import_templates(template_type)
-    template_directory = "#{working_dir}/kit/templates/#{template_type}"
-    return unless Dir.exist?(template_directory)
+    kit_template_dir = "#{working_dir}/kit/templates/#{template_type}"
+    return unless Dir.exist?(kit_template_dir)
+    template_pwd = templates_dirs[template_type]
 
-    FileUtils.cp_r(
-      "#{template_directory}/.",
-      Configuration.send("paths_templates_#{template_type}")
-    )
+    if template_type == 'plugins'
+      FileUtils.cp_r("#{kit_template_dir}/.", template_pwd)
+    else
+      Dir["#{kit_template_dir}/*"].each do |file|
+        return unless File.file?(file)
+        file_name = NamingService.name_file(
+          original_filename: File.basename(file),
+          pathname: template_pwd
+        )
+
+        FileUtils.cp(file, "#{template_pwd}/#{file_name}")
+      end
+    end
```

Comment:
This method is doing much more than what it says in the tin, with different code paths for different argument values, etc.

- Should this be one method or several based on type? or,
- Is there an abstraction that lets you handle all cases? or,
- should the naming piece be extraced?

Reply:
We don't need the `template_type == 'plugins'` path anymore so I removed that and also extracted the naming part 👍